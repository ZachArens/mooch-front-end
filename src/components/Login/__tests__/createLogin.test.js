import React from 'react';
import {fireEvent, cleanup, render} from '@testing-library/react';
import CreateLogin from '../createLogin';

const updateFields = jest.fn();

afterEach(() => {
    cleanup();
});

describe('<CreateLogin />', () => {
    
    test('renders without crashing', () => {
        render(<CreateLogin />);
    });

    test('calls updateFields when when each field is changed', () => {
        const {queryByTestId, debug} = render(<CreateLogin updateFields={updateFields} />);

        fireEvent.change(queryByTestId('verifyEmail'), {
            target: {defaultValue: "test@testemail.com"},
        });

        fireEvent.change(queryByTestId('verifyPassword'), {
            target: {defaultValue: "1234asdfas"},
        });

        fireEvent.change(queryByTestId('fullName'), {
            target: {defaultValue: "Test"},
        });

        fireEvent.change(queryByTestId('streetAddress'), {
            target: {defaultValue: "12 Three Rd"},
        });

        fireEvent.change(queryByTestId('city'), {
            target: {defaultValue: "Testville"},
        });

        fireEvent.change(queryByTestId('st'), {
            target: {defaultValue: "MI"},
        });

        fireEvent.change(queryByTestId('zip'), {
            target: {defaultValue: "12345"},
        });

        fireEvent.change(queryByTestId('phone'), {
            target: {defaultValue: "555-555-5555"},
        });

        expect(updateFields).toHaveBeenCalledTimes(8);

    });

    test('matches snapshot', () => {
        const { container } = render(<CreateLogin />);
        expect(container.firstChild).toMatchSnapshot();
    });
    
});