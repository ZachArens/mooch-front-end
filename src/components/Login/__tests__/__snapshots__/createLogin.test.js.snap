// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateLogin /> matches snapshot 1`] = `
<div>
  <label>
    Verify email and password
    <div>
      <label
        data-testid="emailLabel"
      >
        Email
        <input
          data-testid="verifyEmail"
          name="verifyEmail"
          placeholder="email"
          type="text"
        />
      </label>
      <label
        data-testid="passwordLabel"
      >
        Password
        <input
          data-testid="verifyPassword"
          name="verifyPassword"
          placeholder="********"
          type="password"
        />
      </label>
    </div>
  </label>
  <div
    class="userDetails"
  >
    <label>
      Full Name
      <input
        data-testid="fullName"
        id="fullName"
        name="fullName"
        type="text"
      />
    </label>
    <label>
      Street Address
      <input
        data-testid="streetAddress"
        id="streetAddress"
        name="streetAddress"
        type="text"
      />
    </label>
    <label>
      City
      <input
        data-testid="city"
        id="city"
        name="city"
        type="text"
      />
    </label>
    <label>
      State (2 Letter)
      <input
        data-testid="st"
        id="st"
        name="st"
        type="text"
      />
    </label>
    <label>
      Zip
      <input
        data-testid="zip"
        id="zip"
        name="zip"
        type="text"
      />
    </label>
    <label>
      Phone
      <input
        data-testid="phone"
        id="phone"
        name="phone"
        type="text"
      />
    </label>
  </div>
</div>
`;
